src/v2/coins/eth.ts

```diff
@@ -805,6 +812,182 @@ class Eth extends BaseCoin {
       return response.body;
     }).call(this).asCallback(callback);
   }
+
+
+  /**
+   * Creates the extra parameters needed to build a hop transaction
+   * @param buildParams The original build parameters
+   * @param callback
+   * @returns extra parameters object to merge with the original build parameters object and send to the platform
+   */
+  createHopTransactionParams(buildParams, callback): { hopParams: HopParams, gasLimit: number } {
+    return co(function *() {
+      const wallet = buildParams.wallet;
+      const recipients = buildParams.recipients;
+      const walletPassphrase = buildParams.walletPassphrase;
+
+      const userKeychain = yield this.keychains().get({ id: wallet._wallet.keys[0] });
+      const userPrv = wallet.getUserPrv({ keychain: userKeychain, walletPassphrase });
+      const userPrvBuffer = utxoLib.HDNode.fromBase58(userPrv).getKey().getPrivateKeyBuffer();
+      if (!recipients || !Array.isArray(recipients)) {
+        throw new Error('expecting array of recipients');
+      }
+
+      // Right now we only support 1 recipient
+      if (recipients.length !== 1) {
+        throw new Error('must send to exactly 1 recipient');
+      }
+      const recipientAddress = recipients[0].address;
+      const recipientAmount = recipients[0].amount;
+      const feeEstimateParams = {
+        recipient: recipientAddress,
+        hop: true,
+      };
+      const feeEstimate = yield this.feeEstimate(feeEstimateParams);
+      const gasLimit = feeEstimate.gasLimitEstimate;
+      // shift by 9 because result is in gwei and we want wei
+      const gasPrice = Math.round((feeEstimate.feeEstimate) / gasLimit);
+      const gasPriceMax = gasPrice * 5;
+      // Payment id as  current datetime so its always unique every time we build one
+      const paymentId = (new Date().toISOString());
+      const hopDigest: Buffer = Eth.getHopDigest([recipientAddress, recipientAmount, gasPriceMax, gasLimit, paymentId]);
+
+      const userReqSig = optionalDeps.ethUtil.addHexPrefix(secp256k1.sign(hopDigest, userPrvBuffer).signature.toString('hex'));
```

Comment:
Why are we adding/stripping hex prefixes all over? What's the benefit?

Reply:
whenever its serialized as hex string, all the Eth util stuff (and the HSM) expect the 0x prefix. But, Buffer.from() doesnt work with the hex prefix LOL. So my philosophy with ETH stuff is to always have hex prefix when string and only remove it when shoving it into a buffer